dlc-node/src/main/scala/org/bitcoins/dlc/node/DLCNode.scala

```diff
@@ -35,7 +35,9 @@ case class DLCNode(wallet: DLCWalletApi)(implicit
   }
 
   override def stop(): Future[Unit] = {
-    Future.unit
+    serverBindF.map { case (_, actorRef) =>
+      system.stop(actorRef)
```

Comment:
So i'm not sure if this actually unbinds the underlying port? Just because you stop the actor doesn't mean all resources allocated to that actor are freed. 

Here is how to handle shutdowns correctly inside of the actor: https://doc.akka.io/docs/akka/current/typed/actor-lifecycle.html#stopping-actors

I believe what @rorp said is what you need to send to the underlying `IO(TCP)` actor 

https://github.com/bitcoin-s/bitcoin-s/pull/3402#discussion_r681166523

https://github.com/akka/akka/blob/e127d1916f772de70bd6c144a9e04e1821e3b834/akka-actor/src/main/scala/akka/io/Tcp.scala#L181

Reply:
```scala
var socket = _
def receive = {
  case Tcp.Bound(localAddress) =>
    socket = sender()
  case DLCServer.Disconnect =>
    socket ! Tcp.Unbind
}
```